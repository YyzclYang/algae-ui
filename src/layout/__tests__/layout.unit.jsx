import React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Layout, { Header, Content, Footer, Side } from '../index';

describe('Layout', () => {
  it('渲染 Layout', () => {
    const component = renderer
      .create(
        <Layout>
          <div>Layout</div>
        </Layout>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染包含 Side 的 Layout', () => {
    const component = shallow(
      <Layout>
        <Side>side</Side>
        <Content>content</Content>
      </Layout>
    );

    expect(
      component
        .find('.algae-ui-layout-hasSide')
        .hasClass('algae-ui-layout-hasSide')
    ).toEqual(true);
  });

  it('渲染 Header', () => {
    const component = renderer
      .create(
        <Header>
          <div>Header</div>
        </Header>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染 Content', () => {
    const component = renderer
      .create(
        <Content>
          <div>Content</div>
        </Content>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染 Footer', () => {
    const component = renderer
      .create(
        <Footer>
          <div>Footer</div>
        </Footer>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染 Side', () => {
    const component = renderer
      .create(
        <Side>
          <div>Side</div>
        </Side>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
