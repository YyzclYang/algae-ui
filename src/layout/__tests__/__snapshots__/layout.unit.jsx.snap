// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 渲染 Content 1`] = `
<div
  className="algae-ui-layout-content"
>
  <div>
    Content
  </div>
</div>
`;

exports[`Layout 渲染 Footer 1`] = `
<div
  className="algae-ui-layout-footer"
>
  <div>
    Footer
  </div>
</div>
`;

exports[`Layout 渲染 Header 1`] = `
<div
  className="algae-ui-layout-header"
>
  <div>
    Header
  </div>
</div>
`;

exports[`Layout 渲染 Layout 1`] = `
<div
  className="algae-ui-layout"
>
  <div>
    Layout
  </div>
</div>
`;

exports[`Layout 渲染 Side 1`] = `
<div
  className="algae-ui-layout-side"
>
  <div>
    Side
  </div>
</div>
`;

exports[`Layout 渲染包含 Side 的 Layout 1`] = `
<div
  className="algae-ui-layout"
>
  <div
    className="algae-ui-layout-side"
  >
    <div>
      Side
    </div>
  </div>
</div>
`;
